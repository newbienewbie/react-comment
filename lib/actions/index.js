import Api from '../api';
import {TYPES} from './types';

export {TYPES};


/**
 * 获取首页的次级回复列表
 * @param {String} scope 
 * @param {Number} topicId 
 * @param {Number} page 
 * @param {Number} size 
 * @param {Number} replyPageSize 
 */
export function fetchFirstPageReplyList(scope,topicId,page=1,size=10,replyPageSize=10){

    return dispatch=>{
        return Api.fetchFirstPageReplyList(scope,topicId,page,size,replyPageSize)
            .then(info=>{
                // 触发接收
                const received=fetchFirstPageReplyListReceived(info);
                dispatch(received);
            });
    };
}


/**
 * 获取到首页的回复列表
 * @param {Object} replies 
 */
export function fetchFirstPageReplyListReceived(replies){
    return {
        type:TYPES.fetchFirstPageReplyListReceived,
        replies:replies,
    };
}



/**
 * 获取回复于某个评论（或者某个回复）的回复列表
 * @param {String} scope 
 * @param {Number} topicId 
 * @param {Number} replyUnder 
 * @param {Number} page 
 * @param {Number} size 
 */
export function fetchReplyListOfComment(scope,topicId,replyUnder,page,size){
    return dispatch=>{
        return Api.fetchReplyListUnder(scope,topicId,replyUnder,page,size)
            .then(info=>{
                const {comments,count}=info;
                dispatch(fetchReplyListOfCommentSucceeded(replyUnder,comments,count));
            });
    };
}



/**
 * 请求某个评论（或者回复）的回复列表完成
 * @param {Number} commentId 
 * @param {Array} rows 
 * @param {Number} count 
 */
export function fetchReplyListOfCommentSucceeded(commentId,rows,count){
    return {
        type:TYPES.fetchReplyListOfCommentSucceeded,
        commentId,
        rows,
        count,
    };
}


/**
 * 某个评论的回复翻页
 */
export function replyListPageChanged(commentId,page){
    return {
        type:TYPES.replyListPageChanged,
        commentId,
        page,
    };
}



/**
 * 从服务器获取顶级评论列表
 * @param {String} scope 主题域
 * @param {Number} topicId 主题ID
 * @param {Number} page 评论分页页码
 * @param {Number} size 评论分页每页数量
 */
export function fetchCommentList(scope,topicId,page=1,size=8){
    return dispatch=>{
        return Api.fetchCommentList(scope,topicId,page,size)
            .then(info=>{
                const {comments,count}=info;
                dispatch(fetchCommentListRecevied(comments,count));
                dispatch(commentListPageChanged(page));
            })
    };
}

/**
 * 接收到评论列表
 * @param {Array} comments 
 * @param {Number} count 
 */
export function fetchCommentListRecevied(comments,count){
    return {
        type:TYPES.fetchCommentListRecevied,
        comments,
        count,
    };
}


/**
 * 顶级评论翻页
 */
export function commentListPageChanged(page){
    return {
        type:TYPES.commentListPageChanged,
        page,
    };
}




/**
 * 提交创建评论的请求
 * @param {String} scope 
 * @param {Number} topicId 
 * @param {Number} replyTo 
 * @param {String} content 
 */
export function createComment(scope,topicId,replyTo,content){
    return dispatch=>{
        return Api.create(scope,topicId,replyTo,content)
            .then(comment=>{
                dispatch(createCommentSucceeded(comment));
                const page=1;
                const size=10; // todo: 待修改
                dispatch(fetchCommentList(scope,topicId,replyTo,page,size))
            });
    };
}

/**
 * 创建成功
 * @param {Object} comment 新创建的 comment
 */
export function createCommentSucceeded(comment){
    return {
        type:TYPES.createCommentSucceeded,
        comment,
    };
}


export function createReply(scope,topicId,replyTo,content){
    return dispatch=>{
        return Api.create(scope,topicId,replyTo,content)
            .then(reply=>{
                dispatch(createReplySucceeded(reply));
                // const page=1;
                // const size=10; // todo: 待修改
                // dispatch(fetchReplyListOfComment(scope,topicId,reply.replyUnder,page,size))
                return reply;
            });
    };
}


export function createReplySucceeded(reply){
    return {
        type:TYPES.createReplySucceeded,
        reply,
    };
}


/**
 * 投票反对
 * @param {Number} commentId 
 */
export function downvote(commentId){
    return dispatch=>{
        return Api.downvote(commentId)
            .then(comment=>{
                if(comment){
                    dispatch(downvoteSucceeded(comment));
                }else{
                    dispatch(downvoteFailed(comment));
                }
            });
    };
}



/**
 * 投票反对成功
 */
export function downvoteSucceeded(comment){
    return {
        type:TYPES.downvoteSucceeded,
        comment,
    };
}

/**
 * 投票反对失败
 */
export function downvoteFailed(comment){
    return {
        type:TYPES.downvoteFailed,
        comment,
    };
}



/**
 * 投票赞成处理器
 * @param {Number} commentId 
 */
export function upvote(commentId){
    return dispatch=>{
        return Api.upvote(commentId)
            .then(comment=>{
                if(comment){
                    dispatch( upvoteSucceeded(comment));
                }else{
                    dispatch(upvoteFailed(comment))
                }
            });
    };
}


/**
 * 投票赞成成功
 * @param {Object} comment 
 */
export function upvoteSucceeded(comment){
    return {
        type:TYPES.upvoteSucceeded,
        comment,
    };
}


/**
 * 投票赞成失败
 * @param {Object} comment 
 */
export function upvoteFailed(comment){
    return {
        type:TYPES.upvoteFailed,
    };
}