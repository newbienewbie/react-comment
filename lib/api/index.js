

/**
 * 初始化加载时请求数据
 * @param {String} scope 
 * @param {Number} topicId 
 * @param {Number} replyTo 
 * @param {Number} commentPageSize 
 * @param {Number} replyPageSize 
 */
export function fetchWhenInitialLoading(scope,topicId,replyTo,commentPageSize,replyPageSize){

    const p1 = fetchCommentList(scope, topicId, replyTo, 1, commentPageSize);
    const p2 = fetchReplyList(scope, topicId, 1,commentPageSize, replyPageSize);
    return Promise.all([p1, p2])
        .then(function(results) {
            const _results$ = results[0];
            const comments = _results$.comments;
            const commentsCount = _results$.count;
            const replies = results[1];
            return { comments, commentsCount, replies };
        });
}

/**
 * 从服务器获取评论列表
 * @param {String} scope 主题域
 * @param {Number} topicId 主题ID
 * @param {Number} replyTo 所回复评论的ID，即向哪个评论回复
 * @param {Number} page 评论分页页码
 * @param {Number} size 评论分页每页数量
 */
export function fetchCommentList(scope,topicId,replyTo,page=1,size=8){
    return fetch("/comment/list",{
        method:'post',
        credentials:'same-origin',
        headers:{
            "Content-Type":"application/json",
        },
        body:JSON.stringify({scope, topicId, replyTo,page, size, }),
    })
    .then(resp=>resp.json())
    .then(result=>{
        const comments=result.rows.map(c=>{
            c.author.name=c.author.username;
            return c;
        });
        const count=result.count;
        return {comments,count};
    });
}

/**
 * 获取次级回复列表
 * @param {String} scope 
 * @param {Number} topicId 
 * @param {Number} page 
 * @param {Number} size 
 */
export function fetchReplyList(scope,topicId,page=1,size=10,replyPageSize=10){
    return fetch("/comment/list/reply",{
        method:'post',
        credentials:'same-origin',
        headers:{
            "Content-Type":"application/json",
        },
        body:JSON.stringify({scope, topicId, page, size, replyPageSize}),
    })
    .then(resp=>resp.json());
}


/**
 * 创建新的评论或者回复
 * @param {String} scope 
 * @param {Number} topicId 
 * @param {Number} replyTo 
 * @param {String} content 
 */
export function create(scope,topicId,replyTo,content){
    return fetch(`/comment/new`,{
        method:'post',
        credentials:'same-origin',
        headers:{
            'Content-Type':'application/json',
        },
        body:JSON.stringify({ scope, topicId, replyTo,content})
    })
    .then(resp=>resp.json())
}



/**
 * 投票赞成某评论
 * @param {Number} commentId 
 */
export function upvote(commentId){
    return fetch("/comment/upvote",{
        method:'post',
        credentials:'same-origin',
        headers:{
            "Content-Type":"application/json",
        },
        body:JSON.stringify({id:commentId}),
    })
    .then(resp=>resp.json())
    .then(info=>{
        if(!info){
            return false;
        }else{
            return info[0];
        }
    });
}



/**
 * 投票反对某评论
 * @param {Number} commentId 
 */
export function downvote(commentId){
    return fetch("/comment/downvote",{
        method:'post',
        credentials:'same-origin',
        headers:{
            "Content-Type":"application/json",
        },
        body:JSON.stringify({id:commentId}),
    })
    .then(resp=>resp.json())
    .then(info=>{
        if(!info){
            return false;
        }else{
            return info[0];
        }
    });
}

export default {
    fetchCommentList,
    fetchReplyList,
    create,
    upvote,
    downvote
};