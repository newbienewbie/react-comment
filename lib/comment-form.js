import React from 'react';


/**
 * 评论表单，无状态组件
 */
export const CommentForm=React.createClass({

    getDefaultProps:function(){
        return {
            author:{name:'',email:'',avatarUrl:'#',introduction:'',},
            onSubmit:(content)=>{},
        };
    }, 

    getInitialState:function(){
        return {
            content:''
        };
    },

    render:function(){
        return (<form className="comment-form">
            <div>
                <img src={this.props.author.avatarUrl}/>
                <textarea placeholder='写下你的评论' value={this.state.content} onChange={e=>{this.setState({content:e.target.value});}}/>
            </div>
            <div>
                <button onClick={_=>{this.setState({content:''});}}>取消</button>
                <button onClick={_=>{ 
                    const content=this.state.content.slice(0);
                    this.props.onSubmit(content); 
                    this.setState({content:''}) ;
                }}>
                    发表
                </button>
            </div>
        </form>);
    },
});


export default CommentForm;