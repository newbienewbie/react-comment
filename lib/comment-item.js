import React from 'react';
import {Icon} from 'simple-react-ui';


/**
 * 评论组件，无状态组件，分成三部分
 * * 头部，元信息(作者、统计信息)
 * * 主体：评论
 * * 尾部：操作区
 */
export const CommentItem=React.createClass({

    getDefaultProps:function(){
        return {
            id:'',
            author:{name:'',email:'',avatarUrl:'',introduction:'',},
            content:'',
            upvotes:0,
            downvotes:0,
            createdAt:new Date(),
            updatedAt:new Date(),
            onUpvote:()=>{},
            onDownvote:()=>{},
        };
    },
    
    render() {
        return (<div className='comment-item'>
            {/* header */}
            <div>
                <div>
                    <div>
                        <img src={this.props.author.avatarUrl}/>
                    </div>
                    <div>
                        <span><a href={'#'}>{this.props.author.name}</a></span>
                        <div>{this.props.author.introduction}</div>
                    </div>
                </div>
                <div>
                    <div>{this.props.upvotes}人赞同 
                        <button onClick={(e)=>{
                            e.preventDefault();
                            e.stopPropagation();
                            this.props.onUpvote();
                            return false;
                        }}>
                            赞<Icon.Like/>
                        </button>
                    </div>
                    <div>{this.props.downvotes}人反对 
                        <button onClick={e=>{
                            e.preventDefault();
                            e.stopPropagation();
                            this.props.onDownvote();
                            return false;
                        }}>
                            踩<Icon.Dislike/>
                        </button>
                    </div>
                </div>
            </div>
            {/* body */}
            <div>
                <p>{this.props.content}</p>
                <p>修改于 {this.props.updatedAt.toString()}</p>
            </div>
        </div>);
    }
});


export default CommentItem;
