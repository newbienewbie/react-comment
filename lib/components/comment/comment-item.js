import React from 'react';
import {Icon} from 'simple-react-ui';
import moment from 'moment';
import {Reply} from '../reply';
moment.locale('zh-cn');

/**
 * 评论组件，无状态组件，分成三部分
 * * 头部，元信息(作者、统计信息)
 * * 主体：评论
 * * 尾部：操作区
 */
export class CommentItem extends React.Component{

    constructor(props){
        super(props);
        this.replyRef=null;
    }
 
    render() {
        return (<div className='comment-item'>
            {/* 作者头像 */}
            <div className="comment-author">
                <img src={this.props.author.avatarUrl}/>
            </div>
            {/* body */}
            <div className="comment-body">
                <span><a href={'#'}>{this.props.author.name}</a></span>
                <p>{this.props.content}</p>
                <p>修改于{moment(this.props.updatedAt).fromNow()}</p>
            </div>
            {/* 统计与互动 */}
            <div>
                <div>{this.props.upvotes}人赞同 
                    <button onClick={(e)=>{
                        e.preventDefault();
                        e.stopPropagation();
                        this.props.onUpvote();
                        return false;
                    }}>
                        赞<Icon.Like/>
                    </button>
                </div>
                <div>{this.props.downvotes}人反对 
                    <button onClick={e=>{
                        e.preventDefault();
                        e.stopPropagation();
                        this.props.onDownvote();
                        return false;
                    }}>
                        踩<Icon.Dislike/>
                    </button>
                </div>
                <div>{ this.props.replies.count || 0}条回复 
                    <button onClick={e=>{
                        e.preventDefault();
                        e.stopPropagation();
                        this.setState({replyTo:this.props.replyTo},()=>{
                            // 找到当前顶级评论的次级回复组件的评论框，然后切换焦点
                            this.replyRef.focus();
                        });
                        return false;
                    }}>
                        回复<Icon.Reply/>
                    </button>
                </div>
            </div> 
            <div className="comment-reply">
            </div>
        </div>);
    }
}


CommentItem.defaultProps={
    
    scope:"post",
    topicId:null,

    id:'',
    author:{name:'',email:'',avatarUrl:'',introduction:'',},
    content:'',
    upvotes:0,
    downvotes:0,
    createdAt:new Date(),
    updatedAt:new Date(),

    onUpvote:(commentId)=>{},
    onDownvote:(commentId)=>{},
    replies:{rows:[],count:0},
    replyPageSize:6,
};
   


export default CommentItem;
