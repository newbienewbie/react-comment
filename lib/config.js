
export const config={
    commentPageSize:10,
    replyPageSize:6,
};

export default config;