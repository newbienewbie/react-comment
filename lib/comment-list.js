import React from 'react';
import {CommentItem} from './comment-item.js'; 


/**
 * 评论列表组件，无状态组件，根据传递的属性呈现
 */
export const CommentList=React.createClass({
    getDefaultProps:function(){
        return {
            comments:[
                {id:'',author:{name:'',email:'',avatarUrl:'',introduction:'',},content:'',upvotes:0,downvotes:0,createdAt:'',updatedAt:''},
            ],
            onUpvote:(i)=>{},
            onDownvote:(i)=>{},
        };
    },
    render:function () {
        return (<div>
            {this.props.comments.map((c,i)=>{
                return <CommentItem key={i} 
                    id={c.id} author={c.author} content={c.content}
                    createdAt={c.createdAt} updatedAt={c.updatedAt}
                    onUpvote={()=>{ this.props.onUpvote(c.id); }} 
                    onDownvote={()=>{ this.props.onDownvote(c.id); }} 
                    upvotes={c.upvotes} downvotes={c.downvotes}
                />;
            })}
        </div>);
    }
});


export default CommentList;