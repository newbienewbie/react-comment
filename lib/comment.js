import React from 'react';
import 'whatwg-fetch';
import Pagination from 'simple-react-ui/dist/pagination';
import {CommentForm}  from './comment-form';
import {CommentItem} from './comment-item';
import {configFn}  from './default-fn';


/**
 * Comment 组件，开箱即用，由评论表单、评论列表、分页三部分构成，可以自定义
 * * fetchCommentList
 * * create
 * * onUpvote 
 * * onDownvote
 * 属性
 */
export class Comment extends React.Component{
    

    constructor(props){
        super(props);
        this.state={
            comments:[
                {id:null,author:{name:'',email:'',avatarUrl:'#',introduction:'',},content:'',createdAt:new Date(),updatedAt:new Date(),upvotes:0,downvotes:0,},
            ],
            count:0,
            page:1,
            size:this.props.defaultCommentPageSize,
            /**
             * 各顶级评论的次级回复信息组成的列表——不是数组，是普通Object
             */
            replies:{}, 
        };
    }

    componentDidMount(){
        const {scope,topicId,replyTo}=this.props;
        const p1=this.props.fetchCommentList(scope,topicId,replyTo,1,this.state.size);
        const p2=this.props.fetchReplyList(scope,topicId,1,this.state.size,this.props.defaultReplyPageSize);
        return Promise.all([p1,p2])
            .then(results=>{
                const {comments,count}=results[0];
                let replies=results[1];
                const obj={ comments, count, replies };
                this.setState(obj);
            });
    }


    _handleUpvote(id){
        return this.props.onUpvote(id)
            .then(info=>{
                if(!info){alert('每个人只能表达一次赞或踩');}
                const comment=info;
                console.log(`comment is `,comment.id);
                const comments=this.state.comments.slice();
                comments.forEach(c=>{
                    if(c.id==comment.id){
                        console.log('找到了');
                        c.upvotes=comment.upvotes;
                    }
                })
                console.log(comments);
                this.setState({comments:comments});
            });
    }

    _handleDownVote(id){
        return this.props.onDownvote(id)
            .then(info=>{
                if(!info){alert('每个人只能表达一次赞或踩');}
                const comment=info;
                const comments=this.state.comments.slice();
                comments.forEach(c=>{
                    if(c.id==comment.id){
                        c.downvotes=comment.downvotes;
                    }
                })
                this.setState({comments:comments});
            });
    }

    _getReplies(id){
        let replies=this.state.replies[id];
        if(!replies){
            replies= {rows:[],count:0};
        }
        return replies;
    }

    render() {
        return (<div>
            <CommentForm replyTo={this.props.replyTo} author={{avatarUrl:'#'}} onSubmit={content=>{
                const {scope,topicId,replyTo}=this.props;
                this.props.create(scope,topicId,replyTo,content)
                    .then(info=>{
                        const {page,size}=this.state;
                        return this.props.fetchCommentList(scope,topicId,replyTo,page,size);
                    }).then((result)=>{
                        const {comments,count}=result;
                        this.setState({comments,count});
                    });
            }} />

            {/* comment list */}
            <div>
                {this.state.comments.map((c,i)=>{
                    return <CommentItem key={i} 
                        scope={this.props.scope} topicId={this.props.topicId} replyTo={this.props.replyTo} replyUnder={null}
                        id={c.id} author={c.author} content={c.content}
                        createdAt={c.createdAt} updatedAt={c.updatedAt}
                        upvotes={c.upvotes} downvotes={c.downvotes}
                        replies={this._getReplies(c.id)}
                        onUpvote={()=>{ this._handleUpvote(c.id); }} 
                        onDownvote={(id)=>{this._handleDownVote(c.id); }} 
                    />;
                })}
            </div>
            
            <Pagination current={this.state.page} size={this.state.size} total={this.state.count} 
                onChange={(page)=>{
                    const {scope,topicId,replyTo}=this.props;
                    this.props.fetchCommentList(scope,topicId,replyTo,page,this.state.size)
                        .then(result=>{
                            const {comments,count}=result;
                            this.setState({ comments,count,page });
                        });
                }}
            />
        </div>);
    }
}


// 获取默认配置的若干函数作为组件的默认属性
const {create,fetchCommentList,fetchReplyList,onUpvote,onDownvote}=configFn();

Comment.defaultProps={
    /**
     * 评论组件下的诸多条评论所对应的主题ID
     */
    topicId:null,
    /**
     * 评论组件下的诸多评论所对应的域标识
     */
    scope:'post',
    /**
     * 回复于哪个评论
     */
    replyTo:null,

    /**
     * 顶级评论默认分页大小
     */
    defaultCommentPageSize:10,

    /**
     * 每条评论的次级回复的默认分页大小
     */
    defaultReplyPageSize:6,
    /**
     * 获取评论列表
     */
    fetchCommentList:fetchCommentList,
    /**
     * 获取次级回复列表
     */
    fetchReplyList:fetchReplyList,
    /**
     * 创建新评论
     */
    create:create,
    /**
     * 投票支持某条评论
     */
    onUpvote:onUpvote,
    /**
     * 投票反对某条评论
     */
    onDownvote:onDownvote,
};



export default Comment;