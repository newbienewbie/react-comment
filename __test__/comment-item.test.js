import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ReactTestUtils from 'react-dom/test-utils';
import {CommentItem} from '../lib/comment-item';


describe("test《comment-item.js》",function(){

    const props={
        id:'888',
        author:{name:'webbot',email:'itminus@hotmail.com',avatarUrl:'#',introduction:'sth ',},
        content:'hello,world',
        upvotes:2,
        downvotes:0,
        createdAt:new Date('2017-05-15T12:03:25.742Z'),
        updatedAt:new Date('2017-05-15T12:07:25.742Z'),
        onUpvote:(id)=>{},
        onDownvote:(id)=>{},
    };
    


    it("compare snapshot",()=>{
        const component=renderer.create(<CommentItem 
            id={props.id} author={props.author}  
            content={props.content}
            upvotes={props.upvotes}
            downvotes={props.downvotes}
            createdAt={props.createdAt}
            updatedAt={props.updatedAt}
            onUpvote={props.onUpvote}
            onDownvote={props.onDownvote}
        />);
    
        let tree=component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    
    
    it("simulate click",()=>{
    
        // mock function
        const onUpvote=jest.fn();
        const onDownvote=jest.fn();
    
        const component=ReactTestUtils.renderIntoDocument(<CommentItem
            id={props.id} author={props.author}  
            content={props.content}
            upvotes={props.upvotes}
            downvotes={props.downvotes}
            createdAt={props.createdAt}
            updatedAt={props.updatedAt}
            onUpvote={onUpvote}
            onDownvote={onDownvote}
        />);
    
        // find all the buttons
        const btns=ReactTestUtils.scryRenderedDOMComponentsWithTag(component,"button");
    
        // click upvote button
        ReactTestUtils.Simulate.click(btns[0]);
        expect(onUpvote).toBeCalled();
        ReactTestUtils.Simulate.click(btns[0]);
        expect(onUpvote.mock.calls.length).toBe(2);
    
        // click downvote button
        ReactTestUtils.Simulate.click(btns[1]);
        expect(onDownvote).toBeCalled();
        ReactTestUtils.Simulate.click(btns[1]);
        expect(onDownvote.mock.calls.length).toBe(2);
    
    });

});
    