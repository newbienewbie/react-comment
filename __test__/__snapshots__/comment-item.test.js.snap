exports[`test《comment-item.js》 compare snapshot 1`] = `
<div
  className="comment-item">
  <div>
    <div>
      <div>
        <img
          src="#" />
      </div>
      <div>
        <span>
          <a
            href="#">
            webbot
          </a>
        </span>
        <div>
          sth 
        </div>
      </div>
    </div>
    <div>
      <div>
        2
        人赞同
        <button
          onClick={[Function]}>
          赞
          <svg
            height="16"
            viewBox="0 0 20 18"
            width="13">
            <g>
              <path
                d="M.718 7.024c-.718 0-.718.63-.718.63l.996 9.693c0 .703.718.65.718.65h1.45c.916 0 .847-.65.847-.65V7.793c-.09-.88-.853-.79-.846-.79l-2.446.02zm11.727-.05S13.2 5.396 13.6 2.89C13.765.03 11.55-.6 10.565.53c-1.014 1.232 0 2.056-4.45 5.83C5.336 6.965 5 8.01 5 8.997v6.998c-.016 1.104.49 2 1.99 2h7.586c2.097 0 2.86-1.416 2.86-1.416s2.178-5.402 2.346-5.91c1.047-3.516-1.95-3.704-1.95-3.704l-5.387.007z" />
            </g>
          </svg>
        </button>
      </div>
      <div>
        0
        人反对
        <button
          onClick={[Function]}>
          踩
          <svg
            height="16"
            viewBox="0 0 20 18"
            width="13">
            <g>
              <path
                d="M.718 7.024c-.718 0-.718.63-.718.63l.996 9.693c0 .703.718.65.718.65h1.45c.916 0 .847-.65.847-.65V7.793c-.09-.88-.853-.79-.846-.79l-2.446.02zm11.727-.05S13.2 5.396 13.6 2.89C13.765.03 11.55-.6 10.565.53c-1.014 1.232 0 2.056-4.45 5.83C5.336 6.965 5 8.01 5 8.997v6.998c-.016 1.104.49 2 1.99 2h7.586c2.097 0 2.86-1.416 2.86-1.416s2.178-5.402 2.346-5.91c1.047-3.516-1.95-3.704-1.95-3.704l-5.387.007z"
                transform="scale(1,-1) translate(0,-22)" />
            </g>
          </svg>
        </button>
      </div>
    </div>
  </div>
  <div>
    <p>
      hello,world
    </p>
    <p>
      修改于 
      Mon May 15 2017 20:07:25 GMT+0800 (中国标准时间)
    </p>
  </div>
</div>
`;
